textfsm.py

```diff
@@ -177,7 +177,13 @@ def OnCreateOptions(self):
       self.OnClearAllVar()
 
     def OnAssignVar(self):
-      self._value.append(self.value.value)
+      match = re.match(self.value.regex, self.value.value)
```

Comment:
Calling "re.match" for each input line will get very slow as the regexp may need to be re-compiled with each iteration.
Please add a precompiled regexp to the TextFSMValue object and use that to speed things up.

In addition, rather than always doing a regex match, check to see if the 'groups' attribute of the regexp is non-zero and only then perform the new logic.